h2o-bindings/bin/custom/python/gen_infogram.py

```diff
@@ -0,0 +1,322 @@
+rest_api_version = 3  # type: int
+
+def update_param(name, param):
+    if name == 'infogram_algorithm_params':
+        param['type'] = 'KeyValue'
+        param['default_value'] = None
+        return param
+    return None  # param untouched
+
+def class_extensions():
+    def plot(self, train=True, valid=False, xval=False, figsize=(10, 10), title="Infogram", legend_on=True, server=False):
+        """
+        Plot the infogram.  By default, it will plot the infogram calculated from training dataset.  
+        Note that the frame rel_cmi_frame contains the following columns:
+        - 0: predictor names
+        - 1: admissible 
+        - 2: admissible index
+        - 3: relevance-index or total information
+        - 4: safety-index or net information, normalized from 0 to 1
+        - 5: safety-index or net information not normalized
+        
+        :param self: 
+        :param valid: True if to plot infogram from validation dataset
+        :param xval: True if to plot infogram from cross-validation hold out dataset
+        :return: 
+        """
+
+        
+        plt = get_matplotlib_pyplot(server, raise_if_not_available=True)
+        
+        if train:
+            rel_cmi_frame = self.get_relevance_cmi_frame()            
+            if rel_cmi_frame is None:
+                raise H2OValueError("Cannot locate the H2OFrame containing the infogram data from training dataset.")
+        if valid:
+            rel_cmi_frame_valid = self.get_relevance_cmi_frame(valid=True)
+            if rel_cmi_frame_valid is None:
+                raise H2OValueError("Cannot locate the H2OFrame containing the infogram data from validation dataset.")
+        if xval:
+            rel_cmi_frame_xval = self.get_relevance_cmi_frame(xval=True)
+            if rel_cmi_frame_xval is None:
+                raise H2OValueError("Cannot locate the H2OFrame containing the infogram data from xval holdout dataset.")
+
+        rel_cmi_frame_names = rel_cmi_frame.names
+        x_label = rel_cmi_frame_names[3]
+        y_label = rel_cmi_frame_names[4]
+        ig_x_column = 3
+        ig_y_column = 4
+        index_of_admissible = 1
+        features_column = 0
+        x_thresh = self.actual_params["relevance_threshold"]
+        y_thresh = self.actual_params["cmi_threshold"]
```

Comment:
This fails in the `pyunit_PUBDEV_8258_core_infogram_breast_cancer_cv_validFrame_plot.py` test. 

This seems to fix it:
```diff
index 557af1d606..dd3ebbb0cb 100644
--- a/h2o-py/h2o/estimators/infogram.py
+++ b/h2o-py/h2o/estimators/infogram.py
@@ -861,8 +861,8 @@ class H2OInfogram(H2OEstimator):
         ig_y_column = 4
         index_of_admissible = 1
         features_column = 0
-        x_thresh = self.actual_params["relevance_threshold"]
-        y_thresh = self.actual_params["cmi_threshold"]
+        x_thresh = self.actual_params[x_label + "_threshold"]
+        y_thresh = self.actual_params[y_label + "_threshold"]
 
         xmax=1.1
         ymax=1.1
```

Reply:
@wendycwong fixed this on CV branch.